src/Components/Calculator.js

```diff
@@ -0,0 +1,52 @@
+import React, { useState } from 'react';
+
+const Calculator = () => {
+  const [state, setState] = useState({
+    total: null,
+    next: null,
+    operation: null,
+  });
+
+  function handleClick(buttonName) {
+    const newState = Calculator(state, buttonName);
+    setState(newState);
+  }
+  return (
+    <div className="calculator">
+      <div className="buttons-div">
+        <div className="input-field">{state.next || state.total || '0'}</div>
+      </div>
+      <div className="buttons-div">
+        <button type="button" onClick={() => handleClick('AC')}>AC</button>
+        <button type="button" onClick={() => handleClick('+/-')}>+/-</button>
+        <button type="button" onClick={() => handleClick('%')}>%</button>
+        <button type="button" className="operation" onClick={() => handleClick('÷')}>÷</button>
+      </div>
+      <div className="buttons-div">
+        <button type="button" onClick={() => handleClick('7')}>7</button>
+        <button type="button" onClick={() => handleClick('8')}>8</button>
+        <button type="button" onClick={() => handleClick('9')}>9</button>
+        <button type="button" className="operation" onClick={() => handleClick('x')}>x</button>
+      </div>
+      <div className="buttons-div">
+        <button type="button" onClick={() => handleClick('4')}>4</button>
+        <button type="button" onClick={() => handleClick('5')}>5</button>
+        <button type="button" onClick={() => handleClick('6')}>6</button>
+        <button type="button" className="operation" onClick={() => handleClick('-')}>-</button>
+      </div>
+      <div className="buttons-div">
+        <button type="button" onClick={() => handleClick('1')}>1</button>
+        <button type="button" onClick={() => handleClick('2')}>2</button>
+        <button type="button" onClick={() => handleClick('3')}>3</button>
+        <button type="button" className="operation" onClick={() => handleClick('+')}>+</button>
+      </div>
+      <div className="buttons-div">
+        <button type="button" className="zero" onClick={() => handleClick('0')}>0</button>
+        <button type="button" onClick={() => handleClick('.')}>.</button>
+        <button type="button" className="operation" onClick={() => handleClick('=')}>=</button>
+      </div>
+    </div>
+  );
+};
```

Comment:
- [ ] The `Calculator.js` component should match the following given design. please see the design as implemented it
### Your design
![his](https://github.com/kkmanuu/math-magicians/assets/81727988/34471612-8284-4ccf-bcee-70b42bafdc80)

### Original design
![original](https://github.com/kkmanuu/math-magicians/assets/81727988/f76ae8a7-84f9-4f98-8a76-c83978b61fbc)